### PR TITLE
#160617649 Bookmark articles for reading later

### PR DESCRIPTION
**Pull request by kcharles52**
#### What does this PR do?
- Enables users to bookmark articles to read later
#### Description of Task to be completed?
- Enable authenticated users to bookmark or unbookmark articles
#### How should this be manually tested?
- When you click on an article to view, a sidebar with a bookmark appears. 
- Click on the bookmark icon
- To unbookmark, an article, click on profile > bookmarks
- From the list of bookmarks, click on the remove button below the article you want to unbookmark
#### What are the relevant pivotal tracker stories?
[#160617649 ]

#### Screenshoots
- Single article
screen shot 2018-11-23 at 11 34 41

 - Bookmark icon
screen shot 2018-11-23 at 11 34 57

### CODE REVIEW

**Comment by LehruAngela** [comment]
@kcharles52 The half word "bookm" doesnt look good, you may leave it out and display the icon only. The list of bookmarked articles should look similar to the list of drafts and published articles, @kcharles52 @AmosWels have a sync to decide whether to go with cards or horizontal line distinctions.

File: src/components/bookmark/svg.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    <span onClick={props.handlebookmark} id="bookmark">
+      <svg
+        width={width}
+        style={style}
+        height={height}
+        viewBox={viewBox}
+        className={className}
+      >
+        <path
+          d="M19.385 4h-9.77A2.623 2.623 0 0 0 7 6.615V23.01a1.022 1.022 0 0 0 1.595.847l5.905-4.004 5.905 4.004A1.022 1.022 0 0 0 22 23.011V6.62A2.625 2.625 0 0 0 19.385 4zM21 23l-5.91-3.955-.148-.107a.751.751 0 0 0-.884 0l-.147.107L8 23V6.615C8 5.725 8.725 5 9.615 5h9.77C20.275 5 21 5.725 21 6.615V23z"
```
Comment (LehruAngela) [inline src/components/bookmark/svg.js]: Declare this svg path in a different file and import it here rather than writing the whole path in the component

**Review by LehruAngela — CHANGES_REQUESTED** [review]

**Comment by LehruAngela** [comment]
I expect the screenshot for listing bookmarked articles to be attached on your other PR instead

File: src/components/bookmark/svg.js (hunk 1 above)
Comment (kcharles52) [reply src/components/bookmark/svg.js]: It has been changed

**Comment by LehruAngela** [comment]
Replace the favorites icon with the bookmark icon, and add a hover over it to reflect a descriptive text like 'Bookmark article', also rebase with develop,

**Comment by LehruAngela** [comment]
Look at how @AmosWels implemented his actions(view more, delete )on the drafts/published page, your remove button would look better that way. Please change them.

**Comment by LehruAngela** [comment]
Replace the favorites (heart) icon with your bookmark icon, as favorites is the same as bookmarks @kcharles52

File: src/__tests__/articleTests/createArticleReducer.test.js [hunk 2]
```diff
@@ -19,4 +19,25 @@ describe("create article reducers", () => {
...
+  //   const initialState = {
+  //     count: 0,
+  //     next: null,
+  //     previous: null,
+  //     article: {},
+  //     articles: [],
+  //     results: []
+  //     // onearticle: { read_time: "00:00:00" }
+  //   };
+  //   expect(articlesReducer(initialState, startAction)).toEqual({});
```
Comment (LehruAngela) [inline src/__tests__/articleTests/createArticleReducer.test.js]: Remove all commented code @kcharles52

**Review by LehruAngela — CHANGES_REQUESTED** [review]

**Comment by LehruAngela** [comment]
Rebase with develop, and update the screenshots on the PR notes

File: src/__tests__/articleTests/createArticleReducer.test.js (hunk 2 above)
Comment (kcharles52) [reply src/__tests__/articleTests/createArticleReducer.test.js]: I have removed them. I was trying to help write the tests for Amos

File: src/appUrls/index.js [hunk 3]
```diff
@@ -7,10 +7,12 @@ export const API_URLS = {
...
   PUBLISHED_ARTICLES:`${BASE_URL}/api/me/articles/published/`,
-  DRAFTED_ARTICLES:`${BASE_URL}/api/me/articles/drafts/`
+  DRAFTED_ARTICLES:`${BASE_URL}/api/me/articles/drafts/`,
+  CREATE_BOOKMARK: `${BASE_URL}/api/articles/`,
+  DELETE_BOOKMARK: `${BASE_URL}/api/articles/`
 };
 
 export const LOGIN_URL = `${BASE_URL}/api/users/login/`;
+
+export const svg_url = "M19.385 4h-9.77A2.623 2.623 0 0 0 7 6.615V23.01a1.022 1.022 0 0 0 1.595.847l5.905-4.004 5.905 4.004A1.022 1.022 0 0 0 22 23.011V6.62A2.625 2.625 0 0 0 19.385 4zM21 23l-5.91-3.955-.148-.107a.751.751 0 0 0-.884 0l-.147.107L8 23V6.615C8 5.725 8.725 5 9.615 5h9.77C20.275 5 21 5.725 21 6.615V23z";
```
Comment (LehruAngela) [inline src/appUrls/index.js]: This svg_url should be in the assets folder i.e `src/assets/articleAssets/svgIcons.js`

**Review by LehruAngela — CHANGES_REQUESTED** [review]

File: src/__tests__/articleTests/createArticle.test.js [hunk 4]
```diff
@@ -92,3 +92,20 @@ it("test navbar component", () => {
...
+    preventDefault: () => {
+      jest.fn();
+    }
+  };
+  const component = shallow(<CreateArticle {...props} />);
+  component.instance().handleChange(event);
+  component.instance().handleSelectChange(event);
+  component.instance().handleSubmit(event);
+  component.instance().handleEditorChange();
+  component.instance().categories();
```
Comment (LehruAngela) [inline src/__tests__/articleTests/createArticle.test.js]: expect something to happen i.e   `expect(component.instance().handleChange({ preventDefault }));`

**Review by LehruAngela — CHANGES_REQUESTED** [review]

File: src/appUrls/index.js (hunk 3 above)
Comment (kcharles52) [reply src/appUrls/index.js]: Changed

File: src/__tests__/articleTests/createArticle.test.js (hunk 4 above)
Comment (kcharles52) [reply src/__tests__/articleTests/createArticle.test.js]: Updated the test

**Pull request merged**